reggie.py

```diff
@@ -3238,6 +3252,18 @@ def LoadLevel_NSMBW(self, levelData, areaNum):
             self.scene.addItem(com)
             com.UpdateListItem()
 
+    def OpenPuzzle(self, tilesetIndex = None):
+        if tilesetIndex is not None:
+            if self.objAllTab.isTabEnabled(tilesetIndex):
+                tilesets = [globals_.Area.tileset0, globals_.Area.tileset1, globals_.Area.tileset2, globals_.Area.tileset3]
+                tilesetDir = " " + globals_.gamedef.GetGamePath() + "/Texture/" + tilesets[tilesetIndex] + ".arc"
+                #print(tilesetDir)
+            else: return
+        else:
+            tilesetDir = ""
+        subprocess.Popen("python " + globals_.PuzzleDirectory + "puzzle.py" + tilesetDir)
```

Comment:
Really not too happy with the direct use of `subprocess.Popen`. Also, as it stands, the user needs to enter the path with a trailing directory separator (which is not very clear). Also this has no real error handling (i.e. `puzzle.py` might not actually exist, etc.).